azurerm/resource_arm_batch_pool.go

```diff
@@ -137,34 +137,35 @@ func resourceArmBatchPool() *schema.Resource {
 						"id": {
 							Type:         schema.TypeString,
 							Optional:     true,
+							ForceNew:     true,
```

Comment:
Whats the reasoning behind this change? it looks like this property was here but ignored before?

Reply:
Actually it was ignored because the Batch Pool provider did not support to provide the id of a custom image (which is what this PR brings). The `id` field is only used in that case. I also updated all the other field to be optional as `id` is mutually exclusive with the four other properties.